lib/zendesk_apps_tools/server.rb

```diff
@@ -1,45 +1,83 @@
 require 'sinatra/base'
 require 'zendesk_apps_support/package'
+require 'zendesk_apps_support/installed'
 
 module ZendeskAppsTools
   class Server < Sinatra::Base
     set :protection, :except => :frame_options
-    set :public_folder, proc { "#{settings.root}/assets" }
     last_mtime = Time.new(0)
+    last_domain = nil
 
     get '/app.js' do
       content_type 'text/javascript'
 
-      if File.exists? settings.config
-        curr_mtime = File.stat(settings.config).mtime
-        if curr_mtime > last_mtime
-          settings_helper = ZendeskAppsTools::Settings.new
-          settings.parameters = settings_helper.get_settings_from_file(settings.config, settings.manifest)
-          last_mtime = curr_mtime
+      settings_helper = ZendeskAppsTools::Settings.new
+
+      appsjs = []
+      installations = []
+      order = {}
+
+      settings.apps.each_with_index do |app, index|
+        package = app[:package]
+        app_id = installation_id = -(index+1)
+        app_name = package.manifest_json['name'] || 'Local App'
+
+        appsjs << package.compile_js(
+          app_name: app_name,
+          app_id: app_id,
+          assets_dir: "http://localhost:#{settings.port}/#{app_id}/",
+          locale: params['locale']
+        )
+
+        location = begin
+          locations = package.manifest_json['location']
+          locations = [ locations ] if locations.is_a?(String)
+          locations.is_a?(Hash) ? locations : { zendesk: locations }
         end
+
+        location.each do |host, locations|
+          locations = [ locations ] if locations.is_a?(String)
+          locations.each do |location, index|
+            order[location] ||= []
+            order[location] << app_id
+          end
+        end
+
+        if app[:settings_file_path]
+          curr_mtime = File.stat(app[:settings_file_path]).mtime
+          curr_domain = params['subdomain']
```

Comment:
it's a parameter! app.js?locale=en&subdomain=z3nzibar.

I added this a while ago to the framework.